### ISSUE TITLE
Migrate Pickup - Confusing docs

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1087]
In GitLab by @manuel-g-castro on Jul 13, 2023, 09:20

Hello @dbeltrankyl and @kinow,

### Autosubmit Version

v3.15.0b(-foss-2015a-Python-2.7.9)

### Expid affected

a696

### Summary

I have created an experiment and configure it as I wished. Once that was done, I needed to migrate this experiment to @kinow so he can run it with his user.

This experiment was never run (just inspected), thus there are no files in remote. Either way, I asked @ gbonet for the temp folders in each of the platforms for auto-ecearth3 because it is a mandatory parameter in order to migrate.

### Steps to reproduce

Not really sure.

### What is the current bug behavior?

Autosubmit complains the folder does not exist once the receiving user picks up the offer.

### What is the expected correct behavior?

Autosubmit should stat the empty folders to the receiving user.

### Relevant logs and/or screenshots(if applicable)

```
Copying remote files/dirs on nord3v2
Copying from /gpfs/scratch/bsc32/bsc32371/a696/a696 to /gpfs/scratch/bsc32/bsc32841/a696
Rsync launched 1 times. Can take up to 150 retrials or until all data is transfered
[ERROR] Command rsync --timeout=3600 --bwlimit=20000 -aq --remove-source-files /gpfs/scratch/bsc32/bsc32371/a696/a696 /gpfs/scratch/bsc32/bsc32841 in nord4.bsc.es warning: rsync: link_stat "/gpfs/scratch/bsc32/bsc32371/a696/a696" failed: No such file or directory (2)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1189) [sender=3.1.3]
[eCode=6006]
[ERROR] The files/dirs on /gpfs/scratch/bsc32/bsc32371/a696/a696 cannot be copied to /gpfs/scratch/bsc32/bsc32841/a696.[eCode=6012]
 [CRITICAL] Unable to pickup all platforms, the non-moved files are on the TEMP_DIR
 You can try again with autosubmit a696 -p --onlyremote [eCode=7012]
More info at https://autosubmit.readthedocs.io/en/v3.15.0/faq.html
```

### Any other relevant information(if applicable)

@kinow noticed the double a696/a696 in the path for the folder.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jul 13, 2023, 09:38

Hello @manuel-g-castro ,

The migrate takes place in every platform that the experiment uses in this case:

```
PLATFORM = LOCAL
PLATFORM = nord3v2
PLATFORM = transfer_node
HPCARCH = marenostrum4
```

The temp_dir is not the TMP dir of the experiment, instead, it is a temporal folder where both users can access to it ( also it must be in the same partition)

One location, for dt_machine or marenostrum4 would be:

* /gpfs/scratch/bsc32/bsc32371/temp_migrate
* give to it write group permissions 

transfer_node_bscearth000:

`TEMP_DIR = /esarchive/autosubmit/a696/tmp`

This will do nothing because you're not using the bscearth right now.. but I wouldn't put that path never for the mentioned

marenostrum4

`TEMP_DIR = /gpfs/scratch/bsc32/bsc32371/a696/`

This is what is causing it to be moved into ../a696/a696, it should be placed in an intermediate folder, not the final destination ( as the original user may not have permissions, also this wouldn't change the owner permissions ) 

Basically, what migrate  is doing create a folder in 

TEMP_DIR/$EXPID = /gpfs/scratch/bsc32/bsc32371/a696/a696

Again, one place would be

`* /gpfs/scratch/bsc32/bsc32371/temp_migrate` ( not sure if the original user was bsc32371 or bsc32841

transfer-node

The same issue, also you don't have to specify a path if it is a shared folder, so you could only have it in transfer_node

`TEMP_DIR = ` is valid


nord3v2

Same issue,

I'm not sure where the data is right now, because of the commented.. the logs are in /esarchive/autosubmit/Aslogs

https://autosubmit.readthedocs.io/en/latest/userguide/manage/index.html#how-to-migrate-an-experiment ( check the warnings )

**Comment by kinow** [comment]
In GitLab by @manuel-g-castro on Jul 13, 2023, 09:58

Ahh, Okay. So I misunderstood [from the documentation](https://autosubmit.readthedocs.io/en/latest/userguide/manage/index.html#how-to-migrate-an-experiment). I thought AS needed to know the places in which every temporary file of the experiment is stored for transferring and/or change owner. 

Should we change the documentation (and the configuration parameter) to say something like "intermediate", "transitional", or "pass through" folder? 

On another note, since there are no remote files I could just set the path to `/dev/null/` to meet the requirements of the command, right?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jul 13, 2023, 10:02

> Should we change the documentation (and the configuration parameter) to say something like "intermediate", "transitional", or "pass through" folder?


Maybe.. but for AS 4 as it will be confusing for current users, I believe


> On another note, since there are no remote files I could just set the path to `/dev/null/` to meet the requirements of the command, right?

`TEMP_DIR = `

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jul 13, 2023, 10:39

To change the documentation

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on May 2, 2024, 08:46

In 4.1.7 ( note that the next version is 4.1.6), I want to finish the work on autosubmit migrate ( in merge request since 1 year, I will just cherry-pick the changes into a new branch because the merge will be a pain ) . This issue was assigned to you @kinow , but I can do it

**Comment by kinow** [comment]
Sounds good to me @dbeltrankyl , feel free to assign it to you :thumbsup: Thanks!

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jun 13, 2024, 14:49

Removed milestone since it won't be done soon(I believe) so we don't have to bump it each release
